edx_sysadmin/utils/utility.py

```diff
@@ -21,4 +39,232 @@ def get_course_by_id(course_key, depth=0):
+    ):
+        return True
+    return False
+
+
+def create_user_account(data, build_absolute_uri, use_reg_api=True):
+    """
+    Create User Account through "/user_api/v1/account/registration/" API (if available)
+    or directly through "User" and "UserProfile" models
+
+    Arguments:
+    data (dict) - the params to use while creating user account, it can have "username", "name",
+        "email", "password" and many other registration related fields
+
+    build_absolute_uri (method|request.build_absolute_uri) - a request method to build
+        absolute uri
+
+    use_reg_api (boolean) - used to specify which account creation flow should be followed
+
+    Returns:
+    context (dict) - having context to be passed to templates having information about success and
+        failure of account creation
+    """
+    context = {}
+    if is_registration_api_functional() and use_reg_api:
+        context = make_reg_api_request(
+            build_absolute_uri(reverse("user_api_registration")), data=data
+        )
+    else:
+        context = create_user_through_db_models(data)
+
+    return context
+
+
+def create_user_through_db_models(data):
+    """
+    It Registers User through database models
+
+    Arguments:
+    data (dict) - User account details
+
+    Returns:
+    context (dict) - context to be passed to templates having information about success and
+        failure of account creation
+    """
+    context = {}
+    try:
+        if not User.objects.filter(email=data["email"]).exists():
+            user = User(username=data["username"], email=data["email"], is_active=True)
+            password = normalize_password(data["password"])
+            user.set_password(password)
+            user.save()
+
+            profile = UserProfile(user=user)
+            profile.name = data.get("name")
+            profile.save()
+
+            context["success_message"] = HTML(
+                f"A new account has been registered for user: {data['username']}"
+            )
+        else:
+            context["error_message"] = HTML(
+                f"An account with email: {data['email']} already exists"
+            )
+            return context
+    except Exception as err:  # pylint: disable=broad-except
+        context["error_message"] = HTML(
+            f"Account couldn't be created due to following error: {err}"
+        )
+
+    return context
+
+
+def make_reg_api_request(api_endpoint, data):
+    """
+    Make POST request to "/user_api/v1/account/registration/" API to register User
+
+    Arguments:
+    api_endpoint (url) - An absolute url to registration API
+
+    data (dict) - User account details
+
+    Returns:
+    context (dict) - context to be passed to templates having information about success and
+        failure of account creation
+    """
+    context = {}
+
+    resp = requests.post(api_endpoint, data=data)
+
+    if resp.status_code == 200:
+        context["success_message"] = HTML(
```

Comment:
mark these strings for translations

Reply:
Updated.

Reply:
@HamzaIbnFarooq  see i18n comment below.